objects/src/assets/mod.rs

```diff
@@ -22,18 +22,20 @@ use core::{fmt, ops::Deref};
 /// The methodology for constructing fungible and non-fungible assets is described below.
 ///
 /// # Fungible assets
-/// The first 3 elements of a fungible asset are set to the ID of the faucet which issued the
+/// The first element of a fungible asset are set to the ID of the faucet which issued the
 /// asset. This guarantees the properties described above (the first bit is ONE, and the 32 least
-/// significant bits of the 3rd element are all ZEROs).
+/// significant bits are all ZEROs).
 ///
 /// The last element is set to the amount of the asset. This amount cannot be greater than
 /// 2^63 - 1 and thus requires 63-bits to store.
 ///
+/// The second and third elements are set to ZEROs and are not used.
+///
 /// # Non-fungible assets
 /// The 4 elements of non-fungible assets are computed as follows:
 /// - First the asset data is hashed. This compresses an asset of an arbitrary length to 4 field
-///   elements.
-/// - The result of step 1 is hashed together with the ID of the faucet which issued the asset.
+///   elements: [d0, d1, d2, d3].
+/// - d3 is then replaced with the ID of the faucet which issues the asset: [d0, d1, d2, account_id].
```

Comment:
One question here is which element should be set to the account ID:

- I don't think it should be `d3` as this would be the element which is used as index in the SMT.
- I'm also not sure it should be `d0` as it is "too far" from the most significant bits.

So, I'm thinking maybe it should be `d2`.

Also, we need to set the most significant bit of `d3` to `0` so that there are never collisions between fungible and non-fungible assets.

Reply:
> I'm also not sure it should be `d0` as it is "too far" from the most significant bits.

Actually, I take this back. Maybe setting `d0` to the issuing faucet ID is not a bad idea.

Reply:
What are the main considerations you have in mind when determining the position of the faucet ID?

Reply:
I guess the main consideration is that we don't want to have collisions in the first 64-bits (so that Merkle paths don't all end up at the same depth 64 leaf). This means we don't want to use `d3`.

Beyond that, we just need to have an easy way to make sure there are no collisions for the remaining 3 elements. Here,I don't know if there is any reasons to prefer one element to another. And using `d0` for faucet ID may be a good idea as it is "not in the middle" - but that's also not a very strong reason.